New custom calibration STIS notebook

jotaylor opened the pull request:
I also had to update a few small things in the python scripts, to handle the NB runs.

Rplesha commented on notebooks/ullyses-custom-stis.ipynb:
```diff
@@ -0,0 +1,686 @@
+    "- Manually calculating the number of pixels with high dark rate (DQ=16), using a modified threshold.\n",
+    "- Performing a custom extraction to one or more targets in the long slit. This is done e.g. when the target is especially faint and requires a smaller extraction box, when the target is miscentered in the slit, or there is a companion source in the slit.\n",
+    "\n",
+    "Let's reprocess the SZ-76 data by using the YAML files as input to the ULLYSES code."
```
Make a quick note here that there are plots produced by the function after the code finishes running, but we'll look at those in more detail in the sections below.

Rplesha commented on notebooks/ullyses-custom-stis.ipynb:
```diff
@@ -0,0 +1,686 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A few things to note: the improvement for G230L is minimal to nonexistent, but there is a marked difference for the CCD datasets. For both G430L and G750L, then number of spectral elements marked as having a serious DQ flag has dramatically decreased. This means that the ULLYSES coaddition algorithm will no longer have the large number of dropouts we saw before. For G750L, the NIR fringing h ...
```
remind people that they are looking at the Xs for this purpose

Rplesha reviewed (commented):
I think things look really good! With the exception of the numpy import, I am able to run the notebook fully.

There are a few places where I think a little bit more detail could be helpful that I've noted here, but I think it's really great as it's written now!

jotaylor commented on notebooks/ullyses-custom-stis.ipynb:
```diff
@@ -0,0 +1,686 @@
+    "\n",
+    "from astropy.io import fits\n",
+    "\n",
+    "%matplotlib inline\n",
```
I couldn't ever get this to work on my system... 😢 I don't know what the issue is. I get crazy javascript error every time I run it with inline.

Rplesha replied:
We have just the widget and not inline. Could that be the issue?

jotaylor commented:
@Rplesha thanks for the review!! There are two things I'd like to follow up on. Is the fringing explanation more clear now? I kept the order as is, cause I want to display the plot before talking about what we see in it, but I added more details about what causes it. 

As for matplotlib, I replaced my inline call with widget and whenever I make plots, it breaks my notebook :/ Not sure what to do about it.

Rplesha reviewed (approved):
Yes I think it looks great! The fringing makes a lot more sense now and I can see at the bottom how much it's improved, too.

I'm not too worried about the widget stuff. I think it is useful to be able to zoom into plots, but it isn't a showstopper. Perhaps just having it as a commented out version as an option instead?
